pkg/framework/machines.go

```diff
@@ -139,13 +145,19 @@ func DeleteMachines(client runtimeclient.Client, machines ...*mapiv1beta1.Machin
 func WaitForMachinesDeleted(c client.Client, machines ...*mapiv1beta1.Machine) {
 	Eventually(func() bool {
 		for _, m := range machines {
+			machine := &mapiv1beta1.Machine{}
 			err := c.Get(context.Background(), client.ObjectKey{
 				Name:      m.GetName(),
 				Namespace: m.GetNamespace(),
-			}, &mapiv1beta1.Machine{})
+			}, machine)
 			if !apierrors.IsNotFound(err) {
 				return false // Not deleted, or other error.
 			}
+			// Fail fast if machine entered Failed phase
+			if machine.Status.Phase != nil {
+				Expect(*machine.Status.Phase).NotTo(Equal(MachinePhaseFailed),
```

Comment:
would this break the eventually loop? we still want to WaitForMachinesDeleted

Reply:
The failed machine will never be deleted, as it is not going to reconcile. It should break the loop and report an error, unlike returning an error inside the block.

Reply:
A failed machine still gets reconciled by the deletion logic does it not? The machine failed check is after the deletion timestamp check. As far as I can tell a failed machine can still be deleted, it will likely just be quicker.

https://github.com/openshift/machine-api-operator/blob/487298d74beeadfe564c5a7db6b9e37296b641b9/pkg/controller/machine/controller.go#L265-L268